src/prefect/engine.py

```diff
@@ -1798,7 +1798,12 @@ async def orchestrate_task_run(
                     )
                     terminal_state.state_details.cache_key = cache_key
 
+            # Defer to user to decide whether failure is retriable
+            terminal_state.state_details.retriable = (
+                await _check_task_failure_retriable(task, task_run, terminal_state)
+            )
```

Comment:
Should we check that the terminal state `is_failed()` here first before we bother to change the `retriable` bool?

See line 1793-1799. I don't think we'd need to bother doing this for with completed tasks.

Reply:
Was a bit uncertain between "failed" and "not completed", but figure there's no point for states like cancelled. Added check to only run on *failed* states.

Reply:
Awesome!